pkg/logs/input/docker/launcher.go

```diff
@@ -89,36 +96,25 @@ func (l *Launcher) setup() error {
 	return nil
 }
 
-func (l *Launcher) retrySetupInBackground(shouldRetry bool) error {
+func (l *Launcher) retrySetup(shouldRetry bool) error {
 	var retryStrategy retry.Strategy
 	if shouldRetry {
 		retryStrategy = retry.RetryCount
 	} else {
 		retryStrategy = retry.OneTry
 	}
-	l.initRetry.SetupRetrier(&retry.Config{
+	l.setupRetrier.SetupRetrier(&retry.Config{
 		Name:          "docker Launcher setup",
 		AttemptMethod: func() error { return l.setup() },
 		Strategy:      retryStrategy,
 		RetryCount:    math.MaxInt32,
 		RetryDelay:    30 * time.Second,
```

Comment:
potential improvement idea, not something that needs to be changed now: since the retries now happen synchronously whenever the `Launcher` needs access to the docker client (as opposed to in the background every 30sec), we may want to lower this delay? Or, for even better results, re-work how the AD's docker client and Logs' docker client retries are scheduled (or use the same client).

Rationale: if the auto-discovery's docker client is able to connect and picks up new services, the retry delay of the Logs' docker client here may not be expired yet, and in that case the call to `getCli` would still return an error. Therefore, in this case, we would never tail the logs of the containers that are picked up in this timeframe.

Reply:
Indeed. It makes sense to reduce the retry period since, anyhow, tries will now only be attempted when the auto-discovery generates an event and this can only happen if the docker client of the auto-discovery managed to connect to the docker client.

In this PR, I’ll just reduce the retry period.

But I definitively agree that I should work on not having an extra docker client here!
Let me do that in another PR.